internal/services/pubkey_service.go

```diff
@@ -4,18 +4,26 @@ import (
 	"net/http"
 
 	"github.com/RHEnVision/provisioning-backend/internal/dao"
+	"github.com/RHEnVision/provisioning-backend/internal/db"
 	"github.com/RHEnVision/provisioning-backend/internal/payloads"
 	"github.com/go-chi/render"
 	"github.com/pkg/errors"
 )
 
+var FingerprintReadOnlyError = errors.New("fingerprint is a read-only field")
+
 func CreatePubkey(w http.ResponseWriter, r *http.Request) {
 	payload := &payloads.PubkeyRequest{}
 	if err := render.Bind(r, payload); err != nil {
 		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), err))
 		return
 	}
 
+	// validate payload
+	if payload.Fingerprint != "" {
+		renderError(w, r, payloads.NewInvalidRequestError(r.Context(), FingerprintReadOnlyError))
+	}
+
```

Comment:
We do not allow setting the fingerprint explicitly.

Reply:
is this conflicting with the validations approach? I feel like we are scattering validations across the codebase :thinking:

Reply:
Damn sure I just forgot we have that now!

Reply:
You know what, this check is useless, everytime we create new record it will be overwritten anyways. And we do not allow updating of keys.

Reply:
Removed.